src/main/java/beam/agentsim/events/ReserveRideHailEvent.java

```diff
@@ -0,0 +1,96 @@
+package beam.agentsim.events;
+
+import org.matsim.api.core.v01.Coord;
+import org.matsim.api.core.v01.Id;
+import org.matsim.api.core.v01.events.Event;
+import org.matsim.api.core.v01.population.Person;
+import org.matsim.core.api.internal.HasPersonId;
+import org.matsim.vehicles.Vehicle;
+
+import java.util.Map;
+
+/**
+ * BEAM
+ */
+public class ReserveRideHailEvent extends Event implements HasPersonId {
+    public final static String EVENT_TYPE = "ReserveRideHail";
+    public final static String ATTRIBUTE_PERSON_ID = "person";
+    public final static String ATTRIBUTE_DEPART_TIME = "departTime";
+    public final static String ATTRIBUTE_PICKUP_LOCATION_X = "originX";
+    public final static String ATTRIBUTE_PICKUP_LOCATION_Y = "originY";
+    public final static String ATTRIBUTE_DROPOUT_LOCATION_X = "destinationX";
+    public final static String ATTRIBUTE_DROPOUT_LOCATION_Y = "destinationY";
+
+    private final Id<Person> customerId;
+    private final long departTime;
+    private final double originX;
+    private final double originY;
+    private final double destinationX;
+    private final double destinationY;
+
+    public ReserveRideHailEvent(double time, Id<Person> personId, long departTime, Coord pickUpLocation, Coord dropOutLocation) {
+       this(time,
+               personId,
+               departTime,
+               pickUpLocation.getX(),
+               pickUpLocation.getY(),
+               dropOutLocation.getX(),
+               dropOutLocation.getY());
+    }
+
+
+    public ReserveRideHailEvent(double time, Id<Person> personId, long departTime, double originX,
+                                double originY, double destinationX, double destinationY) {
+        super(time);
+
+        this.customerId = personId;
+        this.departTime = departTime;
+        this.originX = originX;
+        this.originY = originY;
+        this.destinationX = destinationX;
+        this.destinationY = destinationY;
+    }
+
+    private Map<String, String> attr;
+
+    @Override
+    public Map<String, String> getAttributes() {
+        if(attr != null) return attr;
```

Comment:
⚠️ Consider keeping consistent in the formatting:
- Using correct spaces
- Always using curly braces

❗️ Don´t return a mutable reference from a public method. It breaks encapsulation.

## Suggestion

Before:
```Java
public Map<String, String> getAttributes() { 
    if(attr != null) return attr; 
    attr = super.getAttributes(); 
  
    attr.put(ATTRIBUTE_PERSON, personId.toString()); 
    return attr; 
}
```

After:
```Java
public Map<String, String> getAttributes() { 
    if (attr == null) {
        attr = super.getAttributes(); 
        attr.put(ATTRIBUTE_PERSON, personId.toString()); 
    }
    return Collections.unmodifiableMap(attr);
}
```